ethsigner/core/src/integration-test/java/tech/pegasys/ethsigner/jsonrpcproxy/IntegrationTestBase.java

```diff
@@ -238,13 +242,31 @@ void sendPostRequestAndVerifyResponse(
 
   void sendPostRequestAndVerifyResponse(
       final EthSignerRequest request, final EthSignerResponse expectResponse, final String path) {
+    sendPostRequestAndVerifyResponse(request, expectResponse, path, Optional.empty());
+  }
 
-    final Response response =
+  void sendPostRequestAndVerifyResponse(
+      final EthSignerRequest request,
+      final EthSignerResponse expectResponse,
+      final String path,
+      final Optional<Integer> maybeTimeoutInMilliSec) {
+
+    final RequestSpecification requestSpec =
         given()
             .when()
             .body(request.getBody())
-            .headers(RestAssuredConverter.headers(request.getHeaders()))
-            .post(path);
+            .headers(RestAssuredConverter.headers(request.getHeaders()));
+
+    maybeTimeoutInMilliSec.ifPresent(
+        (timeoutInMilliSec) ->
+            requestSpec.config(
+                RestAssured.config()
+                    .httpClient(
+                        HttpClientConfig.httpClientConfig()
+                            .setParam(
+                                CoreConnectionPNames.CONNECTION_TIMEOUT, timeoutInMilliSec))));
```

Comment:
Does the socket timeout also need to be increased?

Reply:
Done